### PR TITLE
[WIP] Fix location of package prefixes

### PR DESCRIPTION
**Pull request by thewtex**
The current implementation had logic errors -- it did not support paths containing a
'.'.

Support the "root package" and duplicate the logic from distutils, which is
documented and expected.

  https://docs.python.org/2/distutils/setupscript.html#listing-whole-packages

Note that this commit starts a new test directory, `unit`.

### CODE REVIEW

**Comment by thewtex** [comment]
Rebased on upstream/master

**Comment by thewtex** [comment]
I will work on improving the coverage...

**Comment by jcfr** [comment]
> I will work on improving the coverage...

Thanks. That would be nice.

File: skbuild/setuptools_wrap.py [hunk 1]
```diff
@@ -194,7 +236,7 @@ def _classify_files(install_paths, package_data, package_prefixes, py_modules,
                 "    Project Root  : {}\n"
                 "    Violating File: {}\n").format(install_root, test_path))
 
-        # peel off the 'skbuild' prefix
+            # peel off the 'skbuild' prefix
```
Comment (opadron) [inline skbuild/setuptools_wrap.py]: Should be aligned with code on the next line.

File: tests/unit/root-package/lib/_hello.cxx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  "Internal \"_hello\" module",
+  -1,
+  hello_methods
+};
+
+PyMODINIT_FUNC PyInit__hello(void)
+{
+  return PyModule_Create(&hello_module_def);
+}
+#endif /* PY_MAJOR_VERSION >= 3 */
```
Comment (opadron) [inline tests/unit/root-package/lib/_hello.cxx]: Could we have a much more stripped down module example?  We should already have this code in the `hello` test project, and I'd rather not duplicate code needlessly.

File: skbuild/setuptools_wrap.py (hunk 1 above)
Comment (thewtex) [inline skbuild/setuptools_wrap.py]: Done

File: tests/unit/root-package/lib/_hello.cxx (hunk 2 above)
Comment (thewtex) [inline tests/unit/root-package/lib/_hello.cxx]: I will update with a spam example.

**Comment by thewtex** [comment]
:-1: Please do not merge. Do possibly to a subtle change in setuptools (?), the wheel generated by the test no longer contains the C extension.

Closing because I don't think it is worth putting more effort into this path. Focusing instead on #124

**Pull request closed without merging**